### PR TITLE
Replace go-octokit with go-github.

### LINKED ISSUE #16: use go-github instead of octokit
octokit is only partially built-out, go-github is more fully-formed. it is also built by interesting people

https://github.com/google/go-github

### PR DESCRIPTION
**Pull request by parkr**
Fixes #16.

### CODE REVIEW

**Comment by parkr** [comment]
My tests suck (unmocked), but OH WELL.

File: github.go [hunk 1]
```diff
@@ -57,22 +77,17 @@ func mergePullRequest(owner, repo, number string) error {
...
-	_, mergeErr := req.Put(map[string]string{}, &merged)
+	commitMsg := fmt.Sprintf("Merge pull request %v", number)
+	_, _, mergeErr := client.PullRequests.Merge(owner, repo, stringToInt(number), commitMsg)
 
 	if mergeErr != nil {
 		if verbose {
 			fmt.Println("Received an error!", mergeErr)
 		}
-		if strings.Contains(mergeErr.Error(), "405 - Pull Request is not mergeable") {
+		if strings.Contains(mergeErr.Error(), "405 Pull Request is not mergeable") {
```
Comment (willnorris) [inline github.go]: this, and other error checks like it where you're relying on an exact error message, seem very fragile.  I'd recommend checking the status code in the response object.

Comment (willnorris) [inline github.go]: Or I'm not sure if the [Error.code](http://godoc.org/github.com/google/go-github/github#Error) would provide what you want... I forget what that value is.  That might be preferable if there are cases where the HTTP response code doesn't uniquely identify exactly what the error was.

**Comment by willnorris** [comment]
> My tests suck (unmocked), but OH WELL.

Mocking and testing strategies are discussed a bit in https://github.com/google/go-github/issues/113.  That's actually exactly how we test things internally at Google that use go-github.

File: github.go (hunk 1 above)
Comment (parkr) [inline github.go]: from the octokit days – you're right, very brittle! i'll use `err.Code` as mentioned.

Comment (parkr) [inline github.go]: interesting

```text
# _/Users/parker/code/merge-pr
./github.go:86: mergeErr.Code undefined (type error has no field or method Code)
./github.go:114: deleteBranchErr.Code undefined (type error has no field or method Code)
```

maybe i'm not actually getting a `github.Error`...

Comment (willnorris) [inline github.go]: you need to type assert it as a github.Error (see http://blog.golang.org/error-handling-and-go)

Comment (parkr) [inline github.go]: mm, yes of course. I don't mind checking the response, so I'll do `res.StatusCode` for now!

**Comment by parkr** [comment]
Problem: `Error deleting the branch: %v Branch not found` :(

**Pull request merged**